LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/client/LabAccount.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities.client;
+
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+/**
+ * An account object model that will used to represent accounts used for testing purposes.
+ */
+@Getter
+@Accessors(prefix = "m")
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
+@EqualsAndHashCode
+public class LabAccount {
```

Comment:
>LabAccount [](http://example.com/codeflow?start=13&length=10)

Why not just expose the UserInfo object, since it contains lot of other useful info about the user

Reply:
The idea is that the lab team can regress our code by changing their model...for instance they might change the name of the UserInfo type...or they may remove it completely...with this new Lab Api module that I am building, I don't want to take a hard dependency on the lab api specific models...and rather wrap them in a custom model...in this case the Account object..so let's say we regenerate the swagger client in the near future then it is okay if it breaks our internal APIs but it shouldn't break any of our public Lab Api helpers...these should remain unchanged...and that's what I am trying to do here...I've added an interface for the Account model here as well: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1406/commits/ce755bb47e43a01913f0ae9fb9e4816e8e72c4b2

Right now I've kept only 3 methods...but we may add a method there in future for instance called as `getHomeDomain` and then we can implement it to return `getUserInfo.getHomeDomain()` but if lab team changes it in the future and moves the property to say LabInfo then we can change the implementation to `getLabInfo.getHomeDomain()`...but the signature of the public API remains unchanged in our tests.

Reply:
I understand your POV. 
My concern was around LabApi returning a lot of other useful info and we seem to trim it down a lot, I think its fine we can just expose what is needed and then add on later.